src/Microsoft.Azure.Relay/HybridConnectionClient.cs

```diff
@@ -204,10 +204,10 @@ public async Task<HybridConnectionStream> CreateConnectionAsync(IDictionary<stri
                 RelayEventSource.Log.ObjectConnected(traceSource, trackingContext);
                 return new WebSocketStream(webSocket.WebSocket, trackingContext);
             }
-            catch (WebSocketException wsException)
+            catch (Exception exception) when (!WebSocketExceptionHelper.IsRelayContract(exception) && !(exception is ArgumentException))
```

Comment:
>!(exception is ArgumentException) [](start = 102, length = 33)

The test which provides invalid HTTP request headers expects ArgumentException to be throw.

We can either allow ArgumentException here or update the unit tests to expect RelayExceptions instead.

Reply:
My personal preference: RelayException.

---
In reply to: [279075094](https://github.com/Azure/azure-relay-dotnet/pull/106#discussion_r279075094) [](ancestors = 279075094)

Reply:
Ok, this makes things more consistent.

---
In reply to: [279868217](https://github.com/Azure/azure-relay-dotnet/pull/106#discussion_r279868217) [](ancestors = 279868217,279075094)